src/cmd/services/m3coordinator/downsample/flush_handler.go

```diff
@@ -142,7 +143,7 @@ func (w *downsamplerFlushHandlerWriter) Write(
 				Timestamp: time.Unix(0, mp.TimeNanos),
 				Value:     mp.Value,
 			}},
-			Unit: mp.StoragePolicy.Resolution().Precision,
+			Unit: xtime.Millisecond,
```

Comment:
This will have pretty poor compresson for things that can be written in with second precision.  

Could we please do at least something here where we try to find the best? i.e. 
```
switch {
case time.Duration(nanos) % time.Second == 0:
  unit = xtime.Second
case time.Duration(nanos) % time.Millisecond == 0:
  unit = xtime.Millisecond
case time.Duration(nanos) % time.Microsecond == 0:
  unit = xtime.Microsecond
default:
  unit = xtime.Nanosecond
}
```

Reply:
Can you update `flush_handler.go` to use the new method you created? `common.SanitizeUnitForM3DB(mp.StoragePolicy.Resolution().Precision)`

Reply:
yeah, forgot about this